nodes/src/nodes/dataset_cobalt/dataset_loader.py

```
@@ -0,0 +1,294 @@
+        Raises:
+            ValueError: If items is empty or not a list.
+        """
+        # Parse JSON string from textarea input (services.json sends a string
+        # when the field type is textarea).
+        if isinstance(items, str):
+            import json
+
+            try:
+                items = json.loads(items)
+            except json.JSONDecodeError as e:
+                raise ValueError(f'Failed to parse inline items as JSON: {e}') from e
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)